### PR TITLE
Make negative value replaced with small relative value.

### PR DESCRIPTION
**Pull request by dklong-csu**
@bangerth this PR is a follow up to your comment on #84 about replacing negative values with a small number _relative_ to the rest of the data. Previously what I had was negative -> 1e-9*norm(x), and now it is negative -> 1e-9*max(x).

### CODE REVIEW

File: statistics.cpp [hunk 1]
```diff
@@ -36,14 +36,16 @@ double Statistics::log_likelihood(const VectorType& data,
     // further and see that instead of zero, a very small number relative to the rest of the
     // concentrations is perhaps more accurate and plays nicely with the logarithms used later.
     // To this end, we calculate the sum of all positive concentrations to get an idea of the
-    // size scale. Then in the pmf calculation, the negative values are changed to be 1e-9*norm.
```
Comment (bangerth) [inline statistics.cpp:39]: I think it was actually never the norm, right?

File: statistics.cpp [hunk 2]
```diff
@@ -36,14 +36,16 @@ double Statistics::log_likelihood(const VectorType& data,
...
+    // size scale. Then in the pmf calculation, the negative values are changed to be 1e-9 times
+    // the max concentration. We could recalculate the norm adding in this adjustment to the
+    // negative values, but it does not make a substantial difference, so that step is skipped.
     double norm = 0.;
+    double max_conc = 0.;
     for (double concentration : distribution) {
       if (concentration > 0)
       {
         norm += concentration;
+        max_conc = (max_conc < concentration ? concentration : max_conc);
```
Comment (bangerth) [inline statistics.cpp:48]:
```suggestion
        max_conc = std::max(max_conc, concentration);
```

File: statistics.cpp [hunk 3]
```diff
@@ -52,7 +54,7 @@ double Statistics::log_likelihood(const VectorType& data,
     {
       if (distribution[i] < 0)
       {
-        pmf[i] = 1e-9;
+        pmf[i] = 1e-9 * max_conc / norm;
```
Comment (bangerth) [inline statistics.cpp:57]: Yes, this looks better to me.

The result here will not be normalized if you happen to have many 1e-9 values. It probably makes no real difference, though.

File: statistics.cpp (hunk 1 above)
Comment (dklong-csu) [reply statistics.cpp:39]: It divided out, so I didn't include it in the calculation. If the concentration was 1e-9*norm, then the normalization step would be 1e-9*norm/norm=1e-9.

File: statistics.cpp (hunk 3 above)
Comment (dklong-csu) [reply statistics.cpp:57]: I thought it didn't make a difference. I could take a pass through the vector first and find the maximum and then replace the substitution in the norm calculation.

Comment (bangerth) [reply statistics.cpp:66]: I agree that it's not likely going to make a difference.  I'll leave it to you.

**Pull request merged**